src/Styleguide/Components/SelectedCareerAchievements.tsx

```diff
@@ -0,0 +1,308 @@
+import {
+  BorderBox,
+  Box,
+  Flex,
+  Link,
+  Sans,
+  Separator,
+  Serif,
+} from "@artsy/palette"
+import { SelectedCareerAchievementsArtistPage_artist } from "__generated__/SelectedCareerAchievementsArtistPage_artist.graphql"
+import { BlueChip } from "Assets/Icons/BlueChip"
+import {
+  hasSections,
+  highestCategory,
+} from "Components/Artist/MarketInsights/MarketInsights"
+import { Modal } from "Components/Modal/Modal"
+import React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+
+export interface SelectedCareerAchievementsProps {
+  artist: SelectedCareerAchievementsArtistPage_artist
+  border?: boolean
+  Container?: (props: { children: JSX.Element }) => JSX.Element
+}
+
+const CATEGORIES = {
+  "blue-chip": "Blue chip",
+  "top-established": "Established",
+  "top-emerging": "Emerging",
+}
+const CATEGORY_LABEL_MAP = {
+  "blue-chip": "Represented by internationally reputable galleries.",
+  "top-established": "Represented by industry leading galleries.",
+  "top-emerging": "Represented by up-and-coming galleries.",
+}
+
+export interface ArtistInsightProps {
+  type: string
+  label: string
+  value?: string
+  entities?: ReadonlyArray<string>
+}
+
+class ArtistInsight extends React.Component<ArtistInsightProps> {
+  state = {
+    expanded: false,
+  }
+
+  renderEntities(entities) {
+    if (entities.length > 1) {
+      if (this.state.expanded) {
+        return (
+          <Sans size="2" verticalAlign="top" color="black60">
+            {entities.join(", ")}.{" "}
+            <Link onClick={() => this.setState({ expanded: false })}>
+              Show less
+            </Link>
+          </Sans>
+        )
+      } else {
+        return (
+          <Sans size="2" verticalAlign="top" color="black60">
+            {entities[0]}, and{" "}
+            <Link onClick={() => this.setState({ expanded: true })}>
+              {entities.length - 1} more
+            </Link>
+          </Sans>
+        )
+      }
+    } else {
+      return (
+        <Sans size="2" verticalAlign="top" color="black60">
+          {entities[0]}
+        </Sans>
+      )
+    }
+  }
+
+  render() {
+    const { label, value, entities } = this.props
+
+    return (
+      <Flex mt={1} alignItems="top" width={260}>
+        <Flex mr={1} flexGrow={0} alignItems="top">
+          <BlueChip />
+        </Flex>
+        <Flex flexDirection="column">
+          <Box>
+            <Sans size="2">{label}</Sans>
+            {value && (
+              <Sans size="2" color="black60">
+                {value}
+              </Sans>
+            )}
+            {entities && entities.length && this.renderEntities(entities)}
```

Comment:
Rather than plucking and passing `entities` into the `renderEntities` method, can it be plucked from within the method itself and thus keep everything co-located?

Reply:
I like it :+1: